apps/admin-gui/src/app/shared/components/hosts-list/hosts-list.component.ts

```diff
@@ -0,0 +1,109 @@
+import {
+  AfterViewInit,
+  Component,
+  EventEmitter,
+  Input,
+  OnChanges,
+  Output,
+  SimpleChanges,
+  ViewChild
+} from '@angular/core';
+import { Host } from '@perun-web-apps/perun/openapi';
+import { SelectionModel } from '@angular/cdk/collections';
+import { MatPaginator, PageEvent } from '@angular/material/paginator';
+import { MatSort } from '@angular/material/sort';
+import { MatTableDataSource } from '@angular/material/table';
+import { TABLE_ITEMS_COUNT_OPTIONS } from '@perun-web-apps/perun/utils';
+
+@Component({
+  selector: 'perun-web-apps-hosts-list',
+  templateUrl: './hosts-list.component.html',
+  styleUrls: ['./hosts-list.component.css']
+})
+export class HostsListComponent implements AfterViewInit, OnChanges {
+
+  constructor() { }
+
+  @ViewChild(MatSort, { static: true }) set matSort(ms: MatSort) {
+    this.sort = ms;
+    this.setDataSource();
+  }
+
+  @Input()
+  hosts: Host[] = [];
+  @Input()
+  hideColumns: string[] = [];
+  @Input()
+  selection = new SelectionModel<Host>(true, []);
+  @Input()
+  filterValue: string;
+  @Input()
+  pageSize = 10;
+  @Input()
+  disableRouting = false;
+
+  @Output()
+  page: EventEmitter<PageEvent> = new EventEmitter<PageEvent>();
+
+  private sort: MatSort;
+
+  displayedColumns: string[] = ['select', 'id', "name"];
+  dataSource: MatTableDataSource<Host>;
+
+  exporting = false;
+
+  @ViewChild(MatPaginator) paginator: MatPaginator;
+  pageSizeOptions = TABLE_ITEMS_COUNT_OPTIONS;
+
+  ngOnChanges(changes: SimpleChanges) {
+    this.displayedColumns = this.displayedColumns.filter(x => !this.hideColumns.includes(x));
+    this.dataSource = new MatTableDataSource<Host>(this.hosts);
+    this.setDataSource();
+    this.dataSource.filter = this.filterValue;
+  }
+
+
+  setDataSource() {
+    if (!!this.dataSource) {
+      this.dataSource.sort = this.sort;
+      this.dataSource.sortingDataAccessor = (item, property) => {
+        switch (property) {
+          case 'name' :{
+            if (item.hostname) {
+              return item.hostname;
+            }
+            return item.createdBy.slice(item.createdBy.lastIndexOf('=') + 1, item.createdBy.length);
```

Comment:
This should be here. What is in the sortingDataAccessor function should just mirroring what the data cell is showing. I guess you probably just copy pasted it and forget to delete this row.